### DISCUSSION TITLE
Applications dependency documentation and managment

### DISCUSSION THREAD

**Discussion by kisst** [modoboa/modoboa#2430] [Ideas]
This might be two issue in one, but as I think they are strongly linked I try to discuss it in one go.
The way to manage `Applications` is a bit hacky, with it's pip installs and config edits, it would be great to have an application manager function built into the admin interface, so when a user is on the on the /core/info page looking at the applications, they would see a full list of the supported / ready applications and the fact that it is installed or not, would be indicated in the `Installed version` column , and if not installed, then provide the option within the admin page to do the installation, or the update.
And as far as I understand some of the applications are mandatory, or at least mandatory when an other application is in use, these dependencies should be documented.
For example I don't need / use the webmail, but weather I can turn it off or not isn't clear, form the docs.

**Comment by tonioo** [comment]
@kisst Managing applications from the admin panel is unfortunately not so easy, and I'd even say it is not a good idea. You can control which extensions you need at installation time (edit the installer config file). Otherwise, you're not supposed to modify the INSTALLED_APPS setting unless you know what you're doing.

**Reply by kisst** [reply]
Might disagree but could be due to lack of i-depth knowledge, however documentation still could ( and imho should ) be done, so the installation can be kept minimal, as I said in the topic starter, I don't need the webmail module at all, but don't know if I can disable it or not, or if I do what other modules might break.

**Reply by tonioo** [reply]
Documentation can always be improved, no problem with that. A another solution would be to add a new flag to the installer just to specify if you want a minimal or a complete setup?

**Reply by kisst** [reply]
I guess that can work too, unsure on the complexity of the dependencies, but your definition of minimal might be different then others, as far as I see you have the following functions / components:
 - E-mail send /receive ( SMTP )
 - User preferences management   ( user web admin, like mail filters, or auto-response  )
 - Domain/ Server/ User management ( web admin, queue management, domains, dns tools, etc ) 
 - Web e-mail client
 - E-mail client access across IMAP 
 - Calendar server
 - Calendar web client
 
 From this list in my minimal def would not include, web e-mail client, nor calendar neither server or client, even could live without auto-response,
 
 I acknowledge that flexibility automaticity means complexity so probably your approach of having a minimal vs complete works better, considering the project maintenance constrains, so to make sure that the def of minimal is the most common, maybe create a pool to capture the least wanted / used parts ?
